File: workspace/lib/xod/core/cast-boolean-to-pulse/any.js
module.exports.setup = function(e) {
  e.context.state = false;
};

module.exports.evaluate = function(e) {
  var state = e.context.state;
  var newValue = e.inputs.IN;

  e.context.state = newValue;

  if (!(newValue === true && state === false)) return {};

  return { OUT: PULSE };
};

File: workspace/lib/xod/core/cast-boolean-to-pulse/any.cpp
struct State {
  bool state = false;
};

{{ GENERATED_CODE }}

void evaluate(NodeId nid) {
    State* state = getState(nid);
    auto newValue = getValue<input_IN>(nid);

    if (newValue == true && state->state == false)
        emitValue<output_OUT>(nid, 1);

    state->state = newValue;
}
